---
title: "DU High throughput sequencing and rare diseases - Introduction to Galaxy"
author: "Stephanie Le Gras$^{1}$ - slegras@igbmc.fr"
date: "2024 December 11th"
output:
  html_document:
    fig_caption: yes
    toc: yes
    toc_depth: 5
    toc_float: yes
    number_sections: yes
  word_document:
    toc: yes
    toc_depth: '2'
  pdf_document:
    fig_caption: yes
    keep_tex: yes
    toc: yes
    toc_depth: 2
bibliography: references.bib    
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, root.dir="~/Documents/Formations/2023/DUDijon")
```

$^1$ GenomEast platform, IGBMC

# - Log in to Galaxy
- Go to the [Galaxy France website](https://usegalaxy.fr/).
- Log in to your account. If you do not have an account, register first.

![][login]

# - History
## - Create a New History
- Create a new history.

![][createnewhistory]

## - Rename the History to “DNA-seq data analysis”
- Change the name of the new history to “DNA-seq data analysis” by clicking the pencil icon next to the history name. 
- Click on “Unnamed history” at the top of the history panel, enter "DNA-seq data analysis,” and click Save.

![][renamehistory]

# - Importing Files from Your Computer to Galaxy
- Download the file **sample.bed.gz** from [this link](https://seafile.igbmc.fr/d/e5ff06e2623344e68a20/) and upload it to Galaxy.
  - Genome: Mouse (mm9)
  - Format: bed

- Ensure you are in the “DNA-seq data analysis” history (switch to it if needed).

## - Upload Steps:
1. Download the file **sample.bed.gz**.
2. Upload it into Galaxy under the “DNA-seq data analysis” history:
   - Click on **Upload Data**.
   - Drag and drop the file into the upload window.
   - Set:
     - Genome: Mouse (mm9)
     - Format: bed

![][importData]

# - Remove a Dataset
- Remove the **sample.bed.gz** dataset from your history by clicking the delete button.

![][removedataset]

- Verify that your history is empty:
  - Click **Show deleted** at the top of the history panel.
  - Permanently delete the file by clicking **Purge All Deleted Content**.
  - Return to the active view by clicking **Show active**.

![][showDeletedDatasets]

# - Running a Tool
- Download the files **CRN-107_11-R1.fastq.gz** and **CRN-107_11-R2.fastq.gz** from [this link](https://seafile.igbmc.fr/d/1adaad8f80394182a784/).
- Import them into your “DNA-seq data analysis” history:
  - Genome: Human Feb. 2009 (GRCh37/hg19) (hg19)
  - Format: `<auto detect>`

## - Upload Steps:
1. Download both fastq files.
2. Upload them into Galaxy:
   - Click **Upload Data**.
   - Drag and drop **CRN-107_11-R1.fastq.gz** and **CRN-107_11-R2.fastq.gz**.
   - Set the genome for both datasets to **Human Feb. 2009 (GRCh37/hg19) (hg19)**.
   - Click **Start**.

![][UploadFastqFiles]

## - Run the tool:

- Use the **FastQC Read Quality Reports** tool to analyze the quality of the datasets:
  - Input files: **CRN-107_11-R1.fastq** and **CRN-107_11-R2.fastq**.
  - Parameters: Default settings.

- What is the quality encoding of the two fastq files?

![][runtool]
![][qualityencoding]

# - Running Tools Without a Workflow
Analyze the CRN-107 data from reads to variant annotation.

## - Tools and Parameters:
1. **Map with BWA-MEM ** map medium and long reads (> 100 bp) against reference genome
    - Using reference genome: Human (Homo sapiens) (b37): hg19
    - Single or Paired-end reads: Paired
    - Select first set of reads: CRN-107_11-R1.fastq.gz
    - Select second set of reads: CRN-107_11-R2.fastq.gz
    - Set read groups information? Set read groups (Picard style)
      - Read group identifier (ID): Auto-assign Yes
      - Read group sample name (SM): Auto-assign Yes
      - Library name (LB): Auto-assign Yes
      - Platform/technology used to produce the reads (PL): ILLUMINA
      - Platform unit (PU): HS026.2
      - Sequencing center that produced the read (CN): Genomeast
      - Description (DS): CRN-107
      - Predicted median insert size (PI): 250
      - Date that run was produced (DT): 2017-12-13
2. **MarkDuplicates** examine aligned records in BAM datasets to locate duplicate molecules.
    - Select SAM/BAM dataset or dataset collection: **output of BWA mem**
    - Select validation stringency: Silent
3. **FreeBayes** bayesian genetic variant detector
    - BAM or CRAM dataset: **output (bam) of markduplicates**
    - Using reference genome: hg19
    - Limit variant calling to a set of regions?
      - Limit by target file
        - Limit analysis to regions in this BED dataset: **CaptureDesign_chr4.bed**
4. **SnpEff eff**: Annotate variants.
    - **Choose SnpEff eff version 4.3+T.galaxy2** (see below)
    ![][changesnpeffversion]
    - Sequence changes (SNPs, MNPs, InDels): **output of FreeBayes (VCF)**
    - Input format: VCF
    - Output format: VCF (only if input is VCF)
    - Genome source: Locally installed snpEff database
      - Genome: 
        - Homo sapiens: hg19
    - Upstream / Downstream length: No upstream / downstream intervals (0 bases)
5. **VCFtoTab-delimited**: Convert VCF data to TAB-delimited format.
    - Select VCF dataset to convert: **output of SnpEff**

- How many variants are called?

# - Create a Workflow Out of an Existing History
1. Extract a workflow from the “DNA-seq data analysis” history:
   - Go to the history menu and select **Extract Workflow**.

![][extractworkflow]

2. Rename the workflow to “DNA-seq data analysis.”

![][editorrunworklow]

# - Edit a Workflow with the Workflow Editor
1. Open the **DNA-seq data analysis** workflow in the editor:
   - Go to **Workflows** (top menu/side bar) and select **Edit**.

![][editworklow]

2. Add the following tools to the workflow:
   - **Samtools flagstat** (compute mapping statistics after BWA mem).
   - **Filter SAM or BAM** (remove low-quality reads with MAPQ < 20).
   - **Samtools flagstat** (compute mapping statistics after filtering).
   - Rename CRN-107_11-R1.fastq.gz box to Read 1 (fastq).
   - Rename CRN-107_11-R2.fastq.gz box to Read 2 (fastq).
   - Rename CaptureDesign_chr4.bed box to Capture Design (bed).

3. Save the workflow.

![][saveworkflow]

# - Run a Workflow
1. Copy the following files to a new history:
   - **CRN-107_11-R1.fastq.gz**
   - **CRN-107_11-R2.fastq.gz**
   - **CaptureDesign_chr4.bed**

2. Run the **DNA-seq data analysis** workflow:
   - Select the appropriate input files and parameters.

![][runworkflow2]
![][runworkflow1]

- How many reads are discarded due to low mapping quality?


[importData]: images/Galaxy/importData.png "importe data"
[removedataset]: images/Galaxy/RemoveDataset.png "remove dataset"
[showDeletedDatasets]: images/Galaxy/showDeletedDatasets.png "show deleted datasets"
[workflowbeforeediting]: images/Galaxy/workflowbeforeediting.png "workflowbeforeediting"
[login]: images/Galaxy/00-login.png "00-login"
[createnewhistory]: images/Galaxy/01-createnewhistory.png "01-createnewhistory"
[qualityencoding]: images/Galaxy/02-qualityencoding.png "02-qualityencoding"
[renamehistory]: images/Galaxy/02-renamehistory.png "02-renamehistory"
[runtool]: images/Galaxy/02-runtool.png "02-runtool"
[changesnpeffversion]: images/Galaxy/ChooseSnpEffversion.png "Choose SnpEff eff version"
[extractworkflow]: images/Galaxy/03-extractworkflow.png "03-extractworkflow"
[LaunchDragAndDrop]: images/Galaxy/03.1-LaunchDragAndDrop.png "03.1-LaunchDragAndDrop"
[UploadFastqFiles]: images/Galaxy/03.2-UploadFastqFiles.png "03.2-UploadFastqFiles"
[editorrunworklow]: images/Galaxy/05-editorrunworklow.png "05-editorrunworklow"
[editworklow]: images/Galaxy/06-editworklow.png "06-editworklow"
[dnaseqworkflow]: images/Galaxy/07-dnaseqworkflow.png "07-dnaseqworkflow"
[dnaseqorderedworkflow]: images/Galaxy/08-dnaseqorderedworkflow.png "08-dnaseqorderedworkflow"
[saveworkflow]: images/Galaxy/10-saveworkflow.png "10-saveworkflow"
[runworkflow1]: images/Galaxy/12-runworkflow.png "12-runworkflow"
[RunWorkflow2]: images/Galaxy/RunWorkflow.png "RunWorkflow"
[workflowafterediting]: images/Galaxy/workflowafterediting.png "workflowafterediting"
